manticore/ethereum.py

```diff
@@ -712,6 +712,11 @@ def _compile(source_code, contract_name):
                     break
 
         assert(name is not None)
+        # capture source code if file handle was passed as arg
+        if isinstance(source_code, file):
```

Comment:
This won't work with Py3 as there is no `file` constant there.

The proper way to check this in py3 is to do:
```python
>>> from io import IOBase
>>> isinstance(open('/tmp/demo2', 'w'), IOBase)
True
```

This however doesn't work in Py2 so we should probably do:
```
try:
    file_type = file  # Python 2
except NameError:
    file_type = IOBase  # Python 3

# ...
if isinstance(source_code, file_type):
```